src/main/java/org/reduct/client/ReductBucketClient.java

```diff
@@ -0,0 +1,105 @@
+package org.reduct.client;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.reduct.client.config.ServerProperties;
+import org.reduct.common.BucketURL;
+import org.reduct.common.exception.ReductException;
+import org.reduct.model.bucket.BucketSettings;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+public class ReductBucketClient extends ReductClient implements BucketClient {
+
+   /**
+    * Create a new bucket client with the given server properties.
+    * NOTE: Client created without access token will not be able to interact with the server if,
+    * authentication is enabled on the server.
+    *
+    * @param serverProperties The server properties to use.
+    */
+   public ReductBucketClient(ServerProperties serverProperties) {
+      this(serverProperties, null);
+   }
+
+   /**
+    * Create a new bucket client with the given server properties and access token.
+    * NOTE: If, authentication is enabled on the server, an access token with full access must be provided
+    * to create a new bucket.
+    *
+    * @param serverProperties The server properties to use.
+    * @param accessToken      The access token to use.
+    */
+   public ReductBucketClient(ServerProperties serverProperties, String accessToken) {
+      this(serverProperties, HttpClient.newHttpClient(), accessToken);
+   }
+
+   ReductBucketClient(ServerProperties serverProperties, HttpClient httpClient, String accessToken) {
+      super(serverProperties, httpClient, new ObjectMapper(), accessToken);
+      objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+   }
+
+   @Override
+   public String createBucket(String bucketName, BucketSettings bucketSettings) {
+      if (bucketName == null || bucketName.isBlank()) {
+         throw new IllegalArgumentException("Bucket name cannot be null or empty");
+      }
+      String requestBody = serializeSettingsOrEmptyJson(bucketSettings);
+      HttpRequest httpRequest = createHttpRequest(bucketName, requestBody);
+      HttpResponse<Void> httpResponse = executeHttpRequest(httpRequest);
+      return handleResponse(bucketName, httpResponse);
+   }
+
+   private String handleResponse(String bucketName, HttpResponse<Void> httpResponse) {
+      switch (httpResponse.statusCode()) {
```

Comment:
I'm not sure if you need to map error messages here.  ReductStore returns a human readble message in the `-x-reduct-error` header.